akkaConstruct/src/main/scala/akka/contrib/construct/ConstructActorParams.scala

```diff
@@ -0,0 +1,36 @@
+package akka.contrib.construct
+
+import akka.actor.{ActorRef, ActorRefFactory}
+
+sealed trait ConstructActorParams {
+
+  def context: ActorRefFactory
+
+  def constructingActor: Option[ActorRef]
+}
+
+case class ConstructInsideActor(context: ActorRefFactory, self: ActorRef) extends ConstructActorParams {
+  override def constructingActor: Option[ActorRef] = Some(self)
+}
+
+case class ConstructOutsideActor(context: ActorRefFactory) extends ConstructActorParams {
+  override def constructingActor: Option[ActorRef] = None
+}
```

Comment:
Inside/Outside seems strange naming convention. Have not heard this before. docs will be useful here.